pallets/did/src/lib.rs

```diff
@@ -334,6 +334,10 @@ pub mod pallet {
 	/// Service endpoints associated with DIDs.
 	///
 	/// It maps from (DID identifier, service ID) to the service details.
+	// SAFETY of Twox64Concat:
+	// The DID Identifier must be registered on chain first. To register a DID
+	// Identifier, you must provide a valid signature for the identifier or
+	// control the origin that corresponds to the identifier.
```

Comment:
How does this play out with cross-chain origins? I know probably migration would be hard here, but it might still be worth it.

Reply:
The DID that creates the endpoint still needs to be a valid accountId. If you would be able to choose an arbitrary accountId you could impersonate other accounts. So I don't think that this will ever be possible.

Reply:
No, but the resulting account ID is given by our location converter. There might be a way for a chain to craft origins which are then converted into something else on our chain. That would be the same as crafting origins on our chain.

Reply:
Creating specific AccountIds on our chain should never be possible? Even if the origin is coming from another chain.

Reply:
Not saying that. But to have an account on our chain you must have a valid private key to generate the signature. On a different chain, "trying" with different accounts might be cheaper, e.g., if, by absurd, you simply create a chain to spam all possible account IDs. This is probably very theoretical, but still possible. All those origins would be converted into some sort of account ID on our chain, but we don't really know what's the "proof of ownership" for that account ID on the source chain. Maybe this is somehow prevented by having XCM fees in place, which is also a mitigation in our case.